R/trackhub.R

```diff
@@ -372,6 +444,78 @@ setMethod("track", "TrackHubGenome", function(object, name, ...) {
     }
 })
 
+setReplaceMethod("track",signature(object = "TrackHubGenome", value = "ANY"),
+                 .exportToTrackHub)
+
+setReplaceMethod("track",
+                 signature(object = "TrackHubGenome", value = "RsamtoolsFile"),
+                 function(object, name, ..., value)
+                 {
+                     if (missing(name))
+                         name <- basename(path(value))
+                     track(object, name) <- URLencode(path(value))
+                     copyResourceToTrackHub(object, URLencode(index(value)))
+                     object
+                 })
+
+setReplaceMethod("track",
+                 signature(object = "TrackHubGenome", value = "RTLFile"),
+                 function(object, name, ..., value)
+                 {
+                     if (missing(name))
+                         name <- basename(path(value))
+                     track(object, name) <- URLencode(path(value))
+                     object
+                 })
+
+setReplaceMethod("track",
```

Comment:
As we discussed, it would be best to avoid side effects (file system modifications) when manipulating this object.